src/peak_finding/1d_list.clj

```diff
@@ -0,0 +1,82 @@
+;; A peak is any element that is larger or equal to both it's neighbours
+;; Items that only have one neighbour (end items) are considered a peak if
+;; they are larger or equal to their neighbour
+;; This algorithm returns the first peak it finds.
+
+(ns peakfinder)
+
+(defn ^:dynamic find-peak
+  "multiple arity recursive function, for finding a peak in a list"
+  ([list] (->> (middle-val list) (item list) (find-peak list)))
+
+  ([list current-item]
+   (if (empty? list) -1
```

Comment:
For some reason, it is considered more "idiomatic" to check the emptiness of a seq with (seq list) rather than (empty? list).  Note they return opposite values!  (seq list) ~= (not (empty? list))